include/dlaf/eigensolver/bt_band_to_tridiag/mc.h

```diff
@@ -168,175 +297,113 @@ void BackTransformationT2B<Backend::MC, Device::CPU, T>::call(Matrix<T, Device::
-
-      const bool affectsTwoRows = i < (mat_hh.nrTiles().rows() - 1);
-
-      // Note: Since the band is of size (mb + 1), in order to tridiagonalize the matrix,
-      // reflectors are of length mb and start with an offset of 1. This means that the application
-      // of the reflector involves multiple tiles.
-      //
-      // e.g. mb = 4
-      // reflectors   matrix
-      //              X X X X
-      // 1 0 0 0      X X X X
-      // a 1 0 0      X X X X
-      // a b 1 0      X X X X
-      //              -------
-      // a b c 1      Y Y Y Y
-      // 0 b c d      Y Y Y Y
-      // 0 0 c d      Y Y Y Y
-      // 0 0 0 d      Y Y Y Y
-      //
-      // From the drawing above, it is possible to see the dashed tile separation between X and Y,
-      // and how the reflectors on the left are going to be applied. In particular, the first row of
-      // the upper tile is not afftected.
-      //
-      // For this reason, we pre-compute the size of the upper and lower (in case there is one) tiles,
-      // together with the size of the tile of the refelctors (v_rows), that as depicted above it
-      // has a different blocksize (no dashed line, it's a single tile)
-      const std::array<SizeType, 2> sizes{
-          mat_hh.tileSize({i, j}).rows() - 1,
-          affectsTwoRows ? mat_hh.tileSize({i + 1, j}).rows() : 0,
-      };
-      const SizeType v_rows = sizes[0] + sizes[1];
-
-      // Note: In general a tile contains a reflector per column, but in case there aren't enough
-      // rows for storing a reflector whose length is greater than 1, then the number is reduced
-      // accordingly. An exception is represented by the last bottom right tile, where the refelctors
-      // are all the remaining ones (i.e. there may be a length 1 reflector in complex cases)
-      const SizeType k_refls = (j != j_last_sweep) ? std::min(v_rows - 1, mb) : (nsweeps - j * mb);
-
-      const matrix::SubTileSpec v_spec{{0, 0}, {v_rows, k_refls}};
-      const matrix::SubTileSpec v_up{{0, 0}, {sizes[0], k_refls}};
-      const matrix::SubTileSpec v_dn{{sizes[0], 0}, {sizes[1], k_refls}};
-
-      if (k_refls < mat_hh.tileSize({i, j}).cols()) {
-        mat_w.setWidth(k_refls);
-        mat_t.setWidth(k_refls);
-        mat_w2.setHeight(k_refls);
+
+      const GlobalElementIndex ij_e(i * b, j * b);
+      const LocalTileIndex ij(dist_hh.localTileIndex(dist_hh.globalTileIndex(ij_e)));
+
+      // Note:  reflector with size = 1 must be ignored, except for the last step of the last sweep
+      //        with complex type
+      const SizeType nrefls = [&]() {
+        const bool allowSize1 = isComplex_v<T> && j == j_last_sweep && step == steps - 1;
+        const GlobalElementSize delta(dist_hh.size().rows() - ij_e.row() - 1,
+                                      std::min(b, dist_hh.size().cols() - ij_e.col()));
+        return std::min(b, std::min(delta.rows() - (allowSize1 ? 0 : 1), delta.cols()));
+      }();
```

Comment:
Why is this wrapped in a lambda?

Reply:
It's a preference of mine. A `[&]() {}` lambda allows to:
- create a scope for variables (generally useful to keep clean the namespace)
- allow to initialize a `const` value in some complex cases (it's not this case, but [here](https://herbsutter.com/2013/04/05/complex-initialization-for-a-const-variable/) there is a good one).

Reply:
In general, I am a bit wary of lambdas with `[&]` in asynchronous code because I have to then check if something with complicated lifetime is used in the lambda and if that may cause an issue. However, that doesn't seem to be the case here.

Reply:
It's a very good point, but it's not the case here as you said.